test/e2e/memcached_test.go

```diff
@@ -88,59 +98,56 @@ func TestMemcached(t *testing.T) {
 	if err != nil {
-		// HACK: dep cannot resolve non-master branches as the base branch for PR's,
-		// so running `dep ensure` will fail when first running
-		// `operator-sdk new ...`. For now we can ignore the first solve failure.
-		// A permanent solution can be implemented once the following is merged:
-		// https://github.com/golang/dep/pull/1658
-		solveFailRe := regexp.MustCompile(`(?m)^[ \t]*Solving failure:.+github\.com/operator-framework/operator-sdk.+:$`)
-		if !solveFailRe.Match(cmdOut) {
-			t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
-		}
+		t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
 	}
 	ctx.AddCleanupFn(func() error { return os.RemoveAll(absProjectPath) })
 
 	if err := os.Chdir(operatorName); err != nil {
 		t.Fatalf("Failed to change to %s directory: (%v)", operatorName, err)
 	}
+
+	// Get repo being tested. If neither TRAVIS_* repo variable is set, we must be
+	// running locally.
 	repo, ok := os.LookupEnv("TRAVIS_PULL_REQUEST_SLUG")
 	if repo == "" {
 		repo, ok = os.LookupEnv("TRAVIS_REPO_SLUG")
 	}
-	if ok && repo != "" && repo != "operator-framework/operator-sdk" {
+	if (ok && repo != "" && repo != "operator-framework/operator-sdk") || !ok {
+		// Get commit being tested. If neither TRAVIS_* variable is set, we must
+		// be running locally, so use local HEAD commit.
 		commitSha, ok := os.LookupEnv("TRAVIS_PULL_REQUEST_SHA")
-		if commitSha == "" {
+		if !ok {
+			commitSha, ok = localCommitSha, true
+		} else if commitSha == "" {
 			commitSha, ok = os.LookupEnv("TRAVIS_COMMIT")
 		}
 		if ok && commitSha != "" {
-			gopkg, err := ioutil.ReadFile("Gopkg.toml")
+			modBytes, err := ioutil.ReadFile("go.mod")
 			if err != nil {
-				t.Fatal(err)
+				t.Fatalf("Failed to read go.mod: %v", err)
 			}
-			// Match against the '#osdk_branch_annotation' used for version substitution
-			// and comment out the current branch.
-			branchRe := regexp.MustCompile("([ ]+)(.+#osdk_branch_annotation)")
-			gopkg = branchRe.ReplaceAll(gopkg, []byte("$1# $2"))
-			versionRe := regexp.MustCompile("([ ]+)(.+#osdk_version_annotation)")
-			gopkg = versionRe.ReplaceAll(gopkg, []byte("$1# $2"))
-			// Plug in the fork to test against so `dep ensure` can resolve dependencies
-			// correctly.
-			gopkgString := string(gopkg)
-			gopkgLoc := strings.LastIndex(gopkgString, "\n  name = \"github.com/operator-framework/operator-sdk\"\n")
-			gopkgString = gopkgString[:gopkgLoc] + "\n  source = \"https://github.com/" + repo + "\"\n  revision = \"" + commitSha + "\"\n" + gopkgString[gopkgLoc+1:]
-			err = ioutil.WriteFile("Gopkg.toml", []byte(gopkgString), fileutil.DefaultFileMode)
+			modFile, err := modfile.Parse("go.mod", modBytes, nil)
 			if err != nil {
-				t.Fatalf("Failed to write updated Gopkg.toml: %v", err)
+				t.Fatalf("Failed to parse go.mod: %v", err)
+			}
+			sdkPath := "github.com/operator-framework/operator-sdk"
+			modFile.AddReplace(sdkPath, "", sdkPath, commitSha)
```

Comment:
A bit weird why [AddReplace()](https://github.com/rogpeppe/go-internal/blob/master/modfile/rule.go#L598-L633) has the signature for returning an error when it doesn't actually return one, but we should check the returned error in case it changes in the future.